producerConsumer.java

```diff
@@ -0,0 +1,105 @@
+package publisherSubscriber;
+
+import java.util.*;
+
+class sharedResource {
```

Comment:
Class names should be Pascal Case, eg: `SharedResource` see: https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html